Yosemite/YosemiteTests/Stores/OrderStoreTests.swift

```diff
@@ -563,6 +563,163 @@ final class OrderStoreTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    func test_optimistic_update_order_customer_note_correctly() {
+        // Given
+        let orderStore = OrderStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+
+        let originalOrder = sampleOrder()
+        let updatedOrder = originalOrder.copy(customerNote: "Updated!")
+
+        orderStore.upsertStoredOrder(readOnlyOrder: originalOrder, in: viewStorage)
+
+        /// Update: Expected Customer note is actually coming from `order.json` (Customer note == "")
+        network.simulateResponse(requestUrlSuffix: "orders/963", filename: "order")
+
+        // When
+        let result: Result<Networking.Order, Error> = waitFor { promise in
+            let action = OrderAction.updateOrderOptimistically(siteID: self.sampleSiteID, order: updatedOrder, fields: [.customerNote]) { result in
+                promise(result)
+            }
+            orderStore.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        let storageOrder = storageManager.viewStorage.loadOrder(siteID: sampleSiteID, orderID: sampleOrderID)
+       XCTAssertEqual(storageOrder?.customerNote, updatedOrder.customerNote)
+    }
+
+    func test_optimistic_update_order_customer_note_reverts_upon_failure() {
+        // Given
+        let orderStore = OrderStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+
+        let originalOrder = sampleOrder() // (Customer note == "")
+        let updatedOrder = originalOrder.copy(customerNote: "Updated!")
+
+        orderStore.upsertStoredOrder(readOnlyOrder: originalOrder, in: viewStorage)
+
+        network.removeAllSimulatedResponses()
+
+        // When
+        let result: Result<Networking.Order, Error> = waitFor { promise in
+            let action = OrderAction.updateOrderOptimistically(siteID: self.sampleSiteID, order: updatedOrder, fields: [.customerNote]) { result in
+                promise(result)
+            }
+            orderStore.onAction(action)
+        }
+
+        // Then
+        XCTAssertFalse(result.isSuccess)
+        let storageOrder = storageManager.viewStorage.loadOrder(siteID: sampleSiteID, orderID: sampleOrderID)
+        XCTAssertEqual(storageOrder?.customerNote, originalOrder.customerNote)
+    }
+
+    func test_optimistic_update_order_customer_note_delete_upon_failure_and_not_exist_locally() {
```

Comment:
I added a new method to cover the scenario when the update fails and the order doesn't exist locally 👍🏼

Reply:
Super nit: let's rename the method to make it clearer:
```suggestion
    func test_optimistic_update_deletes_order_from_storage_upon_failure_if_it_does_not_exist_locally() {
```